eth/test/DFLobby.test.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  DarkForest,
+  IDiamondReadable__factory,
+  IDiamondWritable__factory,
+  IERC165__factory,
+  IERC173__factory,
+} from '@darkforest_eth/contracts/typechain';
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { expect } from 'chai';
+import hre, { ethers } from 'hardhat';
+import { getInterfaceID } from './utils/TestUtils';
+import { createArena, defaultWorldFixture, World } from './utils/TestWorld';
+
+const _INTERFACE_ID_IERC165 = '0x01ffc9a7';
+const _INTERFACE_ID_IERC721 = '0x80ac58cd';
+const _INTERFACE_ID_IERC721METADATA = '0x5b5e139f';
+const _INTERFACE_ID_IERC721ENUMERABLE = '0x780e9d63';
+
+describe('DarkForestLobby', function () {
+  let world: World;
+  let lobby: DarkForest;
+  const initAddress = ethers.constants.AddressZero;
+  const initFunctionCall = '0x';
+
+  async function worldFixture() {
+    const _world = await loadFixture(defaultWorldFixture);
+    const _lobby = await createArena(_world.user1Core, hre.settings.darkforest.initializers);
+    return { _world, _lobby };
+  }
+
+  beforeEach('load fixture', async function () {
+    const { _world, _lobby } = await loadFixture(worldFixture);
+    world = _world;
+    lobby = _lobby;
+  });
+
+  it('Makes a new lobby', async function () {
+    await expect(world.user1Core.createLobby(initAddress, initFunctionCall)).to.emit(
+      world.contract,
+      'LobbyCreated'
+    );
+  });
+  it('Transfers Ownership', async function () {
+    await expect(world.contract.transferOwnership(world.user1.address)).to.emit(
+      world.contract,
+      'OwnershipTransferred'
+    );
+  });
+  it('msg.sender is owner of new Lobby', async function () {
+    expect(await lobby.owner()).to.equal(world.user1.address);
+  });
+  it('new Lobby has correct SolidState Diamond interfaces', async function () {
+    // IDiamondWritable, IDiamondReadable, IERC173, IERC165
+    const erc165 = getInterfaceID(IERC165__factory.createInterface());
+    expect(await lobby.supportsInterface(erc165._hex)).to.be.true;
```

Comment:
```suggestion
    expect(await lobby.supportsInterface(erc165._hex)).to.equal(true);
```
Chai is notorious for having a bad API. If you use a non-function accessor at the end of your assertions, you could theoretically write invalid tests that pass—e.g. `expect(await lobby.supportsInterface(erc165._hex)).to.be.tru;` would pass. Of course, typescript helps us a bit here but we don't do type checking of the tests to avoid chicken-egg problems with typechain.

Always use functions at the end of your assertions and we won't have those potential issues 🎉

Reply:
Good to know!

Reply:
See: [178a0f9](https://github.com/dfdao/game/pull/7/commits/178a0f9d0ceddac98e5504c6e03c3535f49a63a5)